projects/js-packages/publicize-components/src/components/refresh-jetpack-settings/index.js

```diff
@@ -0,0 +1,23 @@
+import { useDispatch } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { SOCIAL_STORE_ID } from '../../social-store';
+
+/**
+ * HOC that refreshes all of the Jetpack Social settings in the store, to be used in class components.
+ *
+ * @param {object} props - The component props.
+ * @param {boolean} props.shouldRefresh - Whether or not to refresh the settings.
+ * @param {object} props.children - The children to render.
+ * @returns { object } The refreshJetpackSocialSettings function.
+ */
+export default function RefreshJetpackSocialSettingsWrapper( { shouldRefresh, children } ) {
+	const refreshOptions = useDispatch( SOCIAL_STORE_ID ).refreshJetpackSocialSettings;
+
+	useEffect( () => {
+		if ( shouldRefresh ) {
+			refreshOptions();
+		}
+	} );
```

Comment:
I don't think this needs `useEffect`. It can simply be `shouldRefresh && refreshOptions()`, which means this component may not even be required.

Reply:
replied https://github.com/Automattic/jetpack/pull/34142#discussion_r1395348927